.github/workflows/integration-test.yml

```diff
@@ -71,8 +71,13 @@ jobs:
           echo "See [Keeping your GitHub Actions and workflows secure Part 1: Preventing pwn requests](https://securitylab.github.com/resources/github-actions-preventing-pwn-requests)." >> $GITHUB_STEP_SUMMARY
           exit 1
 
+      - name: Check merge commit SHA
+        run: echo "github.event.pull_request.merge_commit_sha = ${{ github.event.pull_request.merge_commit_sha }}"
+
       - name: Checkout source
         uses: actions/checkout@v4
+        with:
+          ref: "${{ github.event.pull_request.merge_commit_sha }}"
```

Comment:
I believe we want to check out `github.event.pull_request.head.sha` here, not the merge commit sha:
https://michaelheap.com/access-secrets-from-forks/

Reply:
Yep, just came to that conclusion as well, but based upon the long discussion here: https://github.com/actions/checkout/issues/518

For potentially a bit more robust/safe approach, I'm going to do this (but correct `HEAD^2` with `HEAD~2`): https://github.com/actions/checkout/issues/518#issuecomment-1661941548

Reply:
Gah! Looks like this might be one of those situations where the updated workflow file won't actually be used until it lands on `main`. See https://github.com/nsidc/earthaccess/actions/runs/11692360203/workflow?pr=869

The workflow file there shows my changes, but the actual workflow execution does not. Ugh!

Reply:
That does seem to be how developing CI/CD tends to work: develop -> yolo -> verify -> repeat. :rofl: 

You could try <https://github.com/nektos/act> to see if you can run it locally. I've never taking it for a spin though

Reply:
I use `act` all the time, and everything passes with `act`, but that's because it doesn't fully replicate a github workflow, particularly in this type of scenario. It isn't aware of the PR and simply uses the code sitting locally on your machine, were `act` is running.

Reply:
`act` is great in that it does greatly reduce the need to actually push to the remote to see if CI/CD will work, except when it doesn't, like we have here :-(